model-server/build.gradle.kts

```diff
@@ -184,3 +190,81 @@ spotless {
                 '\n'*/
     }
 }
+
+// OpenAPI integration
+val basePackage = project.group.toString()
+val openAPIgenerationPath = "$buildDir/generated/openapi"
+
+// We let the Gradle OpenAPI generator plugin build data classes and API interfaces based on the provided
+// OpenAPI specification. That way, the code is forced to stay in sync with the API specification.
+openApiGenerate {
+    generatorName.set("kotlin-server")
+    inputSpec.set(layout.projectDirectory.file("../api/public.yaml").toString())
+    outputDir.set(openAPIgenerationPath)
+    packageName.set(basePackage)
+    packageName.set(basePackage)
+    apiPackage.set(basePackage)
+    modelPackage.set(basePackage)
+    // WARNING: there are patched mustache files used!
+    templateDir.set("$projectDir/src/main/resources/openapi/templates")
+    configOptions.set(
+        mapOf(
+            "library" to "ktor",
+            "omitGradleWrapper" to "true",
+            "featureResources" to "true",
+            "featureAutoHead" to "false",
+            "featureCompression" to "false",
+            "featureHSTS" to "false",
+            "featureMetrics" to "false",
+        ),
+    )
+    globalProperties.putAll(
+        mapOf(
+//            "debugOpenAPI" to "true",
+//            "debugModels" to "true",
+//            "debugSupportingFiles" to "true",
+//            "debugOperations" to "true",
+
+//            "models" to "",
+//            "apis" to "",
+//            "supportingFiles" to "",
+//            "apiTests" to "false",
+//            "modelTests" to "false",
+//            "modelDocs" to "false",
+        ),
+    )
+}
+
+// Ensure that the OpenAPI generator runs before starting to compile
+tasks.named("build") {
+    dependsOn("openApiGenerate")
+}
+tasks.named("processResources") {
+    dependsOn("openApiGenerate")
+}
+tasks.named("compileKotlin") {
+    dependsOn("openApiGenerate")
+}
+tasks.named("runKtlintCheckOverMainSourceSet") {
+    dependsOn("openApiGenerate")
+}
```

Comment:
Isn't it enough to let processResources depend on the generation? I though build and compile depend on processResources again?

Reply:
for me gradle complains when i remove the dependencies for `compileKotlin` and `runKtlintCheckOverMainSourceSet`. the `build` dependency can be removed

Reply:
ok